mods/default/lua/environment/grow_bamboo.lua

```diff
@@ -0,0 +1,41 @@
+-- mods/default/lua/environment/grow_bamboo.lua
+-- =============================================
+-- See README.txt for licensing and other information.
+
+-- Bamboo
+function default.grow_bamboo(pos, node)
+	pos.y = pos.y - 1
+	local name = core.get_node(pos).name
```

Comment:
You can just use core.get_node(node.above).name (if I'm not wrong).
And maybe rename the variable to above_name or something similar.